index/scorch/segment/zap/docvalues.go

```diff
@@ -163,7 +161,19 @@ func (di *docValueIterator) getDocValueLocs(docNum uint64) (uint64, uint64) {
 		return di.curChunkHeader[i].DocNum >= docNum
 	})
 	if i < len(di.curChunkHeader) && di.curChunkHeader[i].DocNum == docNum {
-		return di.curChunkHeader[i].DocDvLoc, di.curChunkHeader[i].DocDvLen
+		var start, end uint64
+		if i > 0 {
+			start = di.curChunkHeader[i].DocDvOffset
+		}
+		// single element case
+		if i == 0 && len(di.curChunkHeader) == 1 {
+			end = di.curChunkHeader[i].DocDvOffset
+		} else if i < len(di.curChunkHeader)-1 {
+			end = di.curChunkHeader[i+1].DocDvOffset
+		} else { // for last element
+			end = start + di.curChunkHeader[0].DocDvOffset
+		}
+		return start, end
```

Comment:
this code roughly looks like repeated, duplicated code?  maybe pull it out into a helper func?